File: sourcecode/Python/461.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        # ^ 按位异或运算，相同为0，不同为1；内置函数count()方法用于统计字符串里某个字符出现的次数
        return bin(x ^ y).count('1')

File: Java/461.java
class Solution {
    public int hammingDistance(int x, int y) {
        x = x^y;// 求出有多少位不同
        y=0;
        while(x!=0){// 求1的个数
            y++;
            x = (x-1)&x;
        }
        return y;
    }
}